go/vt/vttablet/tabletserver/vstreamer/main_test.go

```diff
@@ -160,18 +162,30 @@ func expectLog(ctx context.Context, t *testing.T, input any, ch <-chan []*binlog
 				}
 				for _, ev := range allevs {
 					// Ignore spurious heartbeats that can happen on slow machines.
-					if ev.Type == binlogdatapb.VEventType_HEARTBEAT {
+					if ev.Throttled || ev.Type == binlogdatapb.VEventType_HEARTBEAT {
 						continue
 					}
-					if ev.Throttled {
-						continue
+					switch ev.Type {
+					case binlogdatapb.VEventType_OTHER:
+						if strings.Contains(strings.ToLower(ev.Gtid), "copy start") {
+							inCopyPhase = true
+						}
+					case binlogdatapb.VEventType_FIELD:
+						ev.FieldEvent.EnumSetStringValues = inCopyPhase
+					case binlogdatapb.VEventType_COPY_COMPLETED:
+						inCopyPhase = false
+					}
+					if strings.Contains(ev.String(), "enum(") || strings.Contains(ev.String(), "set(") {
```

Comment:
I'm not sure how `ev` is encoded. I am wondering whether `Contains( ... "enum(")` is safe, or is there a more formal approach?

Reply:
These are unit tests where we're building the events ourselves. These values are the TableMap event's column value which will always be enum('va1,val2...') and set('va1,val2...')

Reply:
I made some changes here that hopefully make it more clear and feel better 🙂 https://github.com/vitessio/vitess/pull/15723/commits/65106311c63d310cd254ad14cfacf69750127bdc